src/llm/functions.rs

```diff
@@ -1,49 +1,82 @@
+use async_trait::async_trait;
+use datafusion::arrow::array::{ArrayRef, RecordBatch};
 use datafusion::arrow::datatypes::DataType;
 use datafusion::common::internal_err;
 use datafusion::common::Result;
-use datafusion::logical_expr::{
-    ColumnarValue, ScalarUDFImpl, Signature, TypeSignature, Volatility,
-};
+use datafusion::logical_expr::{ColumnarValue, ScalarUDF, ScalarUDFImpl, Signature};
 use std::any::Any;
+use std::fmt::Debug;
+use std::sync::Arc;
 
-/// A scalar UDF that will be bypassed when planning logical plan.
-/// This is used to register the remote function to the context. The function should not be
-/// invoked by DataFusion. It's only used to generate the logical plan and unparsed them to SQL.
+/// A scalar UDF that can invoke using async methods
```

Comment:
Here is the the new API. At a high level it is meant to mimic `ScalarUDFImpl` except that it has a `async` invoke function